state/accountsDB_test.go

```diff
@@ -2372,15 +2373,19 @@ func TestAccountsDB_NewAccountsDbStartsSnapshotAfterRestart(t *testing.T) {
 					return true
 				},
 				TakeSnapshotCalled: func(_ []byte, _ []byte, _ chan core.KeyValueHolder, _ common.SnapshotStatisticsHandler, _ uint32) {
+					mutex.Lock()
 					takeSnapshotCalled = true
```

Comment:
could have used an atomic.Flag. Can be left as it is.

Reply:
Done.